vita3k/gui/src/controllers_dialog.cpp

```diff
@@ -173,6 +173,36 @@ static void add_bind_to_table(GuiState &gui, EmuEnvState &emuenv, const SDL_Game
     ImGui::PopID();
 }
 
+static void swap_controller_ports(CtrlState &state, int source_port, int dest_port) {
+    // Check if the ports are valid
+    if (source_port < 1 || source_port > SCE_CTRL_MAX_WIRELESS_NUM || dest_port < 1 || dest_port > SCE_CTRL_MAX_WIRELESS_NUM) {
+        LOG_ERROR("Ports are not valid.");
+        return;
+    }
+
+    // Find the controllers corresponding to the source and destination ports
+    auto source_controller_it = std::find_if(state.controllers.begin(), state.controllers.end(),
+        [source_port](const auto &pair) { return pair.second.port == source_port; });
+    auto dest_controller_it = std::find_if(state.controllers.begin(), state.controllers.end(),
+        [dest_port](const auto &pair) { return pair.second.port == dest_port; });
+
+    // Check that both controllers exist
+    if (source_controller_it == state.controllers.end() || dest_controller_it == state.controllers.end()) {
+        LOG_ERROR("Unable to find one or more controllers on the specified ports.");
+        return;
+    }
+
+    LOG_INFO("Controller on source port {} {} swapped with controller on destination port {} {}", source_port, state.controllers_name[source_port - 1], dest_port, state.controllers_name[dest_port - 1]);
+
+    // Swap controllers port and player index
+    SDL_GameControllerSetPlayerIndex(source_controller_it->second.controller.get(), dest_port - 1);
```

Comment:
Shouldn't `SDL_GameControllerSetPlayerIndex` also be called for dest_controller?

Reply:
is swaped alone when you moving one player index

Reply:
so if you have player 1 2, swap done 2 1 alone

but if have more 2 controller i no know

Reply:
It should work because internally SDL re-assigns the first available index to the other controller, however this relies on the internal behavior of SDL and not on the public API documentation, so I think adding a
`SDL_GameControllerSetPlayerIndex(dest_controller_it ->second.controller.get(), source_port ); `
doesn't hurt

Reply:
ok, done, in first version i have added it, but removed after see is not needed
player index tecnicaly is only important for Playstation controller with it indicate position of player